examples/11_hens/utilities.py

```diff
@@ -0,0 +1,1050 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import secrets
+from collections import OrderedDict
+from concurrent.futures import Future, ThreadPoolExecutor
+from functools import partial
+
+import hydra
+import numpy as np
+import pandas as pd
+import torch
+import xarray as xr
+from loguru import logger
+from omegaconf import DictConfig, open_dict
+from physicsnemo.distributed import DistributedManager
+from reproduce_utilities import ensure_all_torch_seeds_are_unique
+
+from earth2studio.data import DataSource
+from earth2studio.io import IOBackend, KVBackend, XarrayBackend
+from earth2studio.models.auto import Package
+from earth2studio.models.dx import TCTrackerWuDuan
+from earth2studio.models.px import PrognosticModel
+from earth2studio.perturbation import Perturbation
+from earth2studio.utils.time import to_time_array
+
+
+def initialise_perturbation(
+    model: PrognosticModel,
+    data: DataSource,
+    start_time: np.ndarray[np.datetime64],
+    cfg: DictConfig,
+) -> Perturbation:
+    """
+    Initialise perturbation method. Some methods need to be initialized with model, data, start_time, etc.
+    which can not always be defined in the config requireing partial instantiation.
+
+    Parameters
+    ----------
+    model : PrognosticModel
+        forecast model.
+    data : DataSource
+        Data source from which to obtain ICs
+    start_time : np.ndarray[np.datetime64]
+        IC times
+    cfg : DictConfig
+        Hydra config object
+
+    Returns
+    -------
+    Perturbation
+        Perturbation method.
+    """
+    perturbation = hydra.utils.instantiate(cfg.perturbation)
+
+    if isinstance(perturbation, partial):  # inform about model, IC etc
+        if perturbation.func.__name__ == "HENSPerturbation":
+            perturbation = perturbation(model=model, start_time=start_time, data=data)
+        elif perturbation.func.__name__ == "BredVector":
+            perturbation = perturbation(model=model)
+        else:
+            raise ValueError(
+                f"perturbation method {perturbation.func.__name__} not implemented for partial instantiation"
+            )
+
+    return perturbation
+
+
+def build_package_list(cfg: DictConfig) -> list[str]:
```

Comment:
If I understand this correctly right now you have people wgetting the whole checkpoint repo, consider updating this to using the standard way to access the checkpoint on a as needed basis.

I.e. just manage a constant list of each one of these folders and then just build the package object based on the number of models requested

https://nvidia.github.io/earth2studio/examples/11_huge_ensembles.html#set-up